distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Duration.enso

```diff
@@ -87,27 +88,27 @@ type Duration
              example_add = (Duration.new minutes=3) + (Duration.new seconds=6)
 
        > Example
-         Add 12 hours to a duration of a month.
+         Add 12 hours to a duration of 30 minutes.
 
              import Standard.Base.Data.Time.Duration
 
-             example_add = 1.month + (Duration.new hours=12)
+             example_add = (Duration.new minutes=30) + (Duration.new hours=12)
     + : Duration -> Duration ! Time_Error
     + self that =
-        Panic.catch ArithmeticException (self.plus_builtin that) (err-> Error.throw (Time_Error_Data err.getMessage))
+        case that of
+            _ : Duration ->
+                Panic.catch ArithmeticException (self.plus_builtin that) err->
+                    Error.throw (Time_Error_Data err.getMessage)
+            _ : Period.Period ->
+                Error.throw Time_Error_Data "Adding Period to Duration is not supported"
+            _ ->
+                Error.throw Illegal_State_Error_Data "Illegal argument `that`"
```

Comment:
```suggestion
                Error.throw Illegal_Argument_Error_Data "Illegal argument `that`"
```

Let's be consistent with the errors. In other places it was illegal argument, so why make it illegal state here?